File: sandbox/exercism/ruby/2019/bob.rb
class Bob
  def self.hey(remark)
    remark = Remark.new(remark)

    return "Calm down, I know what I'm doing!" if remark.yelling? && remark.asking?
    return 'Whoa, chill out!'                  if remark.yelling?
    return 'Sure.'                             if remark.asking?
    return 'Fine. Be that way!'                if remark.speaking_nothing?

    'Whatever.'
  end
end

class Remark
  attr_reader :remark

  def initialize(remark)
    @remark = remark.strip
  end

  def yelling?
    remark.match?(/[A-Z]+/) && remark == remark.upcase
  end

  def asking?
    remark.end_with?('?')
  end

  def speaking_nothing?
    remark.empty?
  end
end

File: sandbox/exercism/go/2025/bob.go
// https://exercism.org/tracks/go/exercises/bob

package bob

import (
	"regexp"
	"strings"
)

func isAsking(remark string) bool {
	re := regexp.MustCompile(`\?$`)
	return re.MatchString(remark)
}

func isYelling(remark string) bool {
	re := regexp.MustCompile(`[A-Z]+`)
	return re.MatchString(remark) && strings.ToUpper(remark) == remark
}

func isSpeakingNothing(remark string) bool {
	return remark == ""
}

func Hey(remark string) string {
	r := strings.TrimSpace(remark)

	switch {
	case isSpeakingNothing(r): return "Fine. Be that way!"
	case isYelling(r) && isAsking(r): return "Calm down, I know what I'm doing!"
	case isAsking(r): return "Sure."
	case isYelling(r): return "Whoa, chill out!"
	default: return "Whatever."
	}
}
